exonum-java-binding/qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -114,356 +88,394 @@ void tearDown() {
-      Fork view = db.createFork(cleaner);
-      Optional<String> initialConfiguration = service.initialize(view);
-
-      // Check the configuration.
-      assertThat(initialConfiguration).hasValue(INITIAL_SERVICE_CONFIGURATION);
-
-      // Check that both the default and afterCommit counters were created.
-      QaSchema schema = new QaSchema(view);
-      MapIndex<HashCode, Long> counters = schema.counters();
-      MapIndex<HashCode, String> counterNames = schema.counterNames();
-
-      HashCode defaultCounterId = sha256().hashString(DEFAULT_COUNTER_NAME, UTF_8);
-      HashCode afterCommitCounterId = sha256().hashString(AFTER_COMMIT_COUNTER_NAME, UTF_8);
-
-      assertThat(counters.get(defaultCounterId)).isEqualTo(0L);
-      assertThat(counterNames.get(defaultCounterId)).isEqualTo(DEFAULT_COUNTER_NAME);
-      assertThat(counters.get(afterCommitCounterId)).isEqualTo(0L);
-      assertThat(counterNames.get(afterCommitCounterId)).isEqualTo(AFTER_COMMIT_COUNTER_NAME);
+  void initialize() {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      testKit.withSnapshot((view) -> {
+        // TODO: https://jira.bf.local/browse/ECR-2683 is needed for service configuration to be
+        //  checked
+
+        // Check that both the default and afterCommit counters were created.
+        QaSchema schema = new QaSchema(view);
+        MapIndex<HashCode, Long> counters = schema.counters();
+        MapIndex<HashCode, String> counterNames = schema.counterNames();
+
+        HashCode defaultCounterId = sha256().hashString(DEFAULT_COUNTER_NAME, UTF_8);
+        HashCode afterCommitCounterId = sha256().hashString(AFTER_COMMIT_COUNTER_NAME, UTF_8);
+
+        assertThat(counters.get(defaultCounterId)).isEqualTo(0L);
+        assertThat(counterNames.get(defaultCounterId)).isEqualTo(DEFAULT_COUNTER_NAME);
+        assertThat(counters.get(afterCommitCounterId)).isEqualTo(0L);
+        assertThat(counterNames.get(afterCommitCounterId)).isEqualTo(AFTER_COMMIT_COUNTER_NAME);
+        return null;
+      });
     }
   }
 
   @Test
-  void submitCreateCounter() throws Exception {
-    setServiceNode(node);
-
-    when(node.submitTransaction(any(RawTransaction.class))).thenReturn(DEFAULT_TX_MESSAGE_HASH);
-
-    String counterName = "bids";
-    HashCode txHash = service.submitCreateCounter(counterName);
-
-    CreateCounterTx expectedTx = new CreateCounterTx(counterName);
-    RawTransaction expectedRawTx = CreateCounterTx.converter().toRawTransaction(expectedTx);
-
-    assertThat(txHash).isEqualTo(DEFAULT_TX_MESSAGE_HASH);
-    verify(node).submitTransaction(eq(expectedRawTx));
+  void submitCreateCounter() {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      String counterName = "bids";
+      service.submitCreateCounter(counterName);
+      testKit.createBlock();
+
+      testKit.withSnapshot((view) -> {
```

Comment:
(here and below) Actually, these tests now verify the results
not only of submitting a transaction, but of its execution. They became more complex.
As the results of an execution are verified in transaction tests, I would consider simplifying
these.

Also, how can we verify that a certain transaction was submitted with the current testkit API?
Would it be simpler? If not, how can it be simplified?